segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorage.java

```diff
@@ -748,19 +773,19 @@ public void report() {
     CompletableFuture<Void> updateStorageStats() {
         return chunkStorage.getUsedSpace()
                 .thenAcceptAsync(used -> {
-                    storageUsed.set(used);
+                    healthTracker.setStorageUsed(used);
                     boolean isFull = used >= config.getMaxSafeStorageSize();
                     if (isFull) {
-                        if (!isStorageFull.get()) {
+                        if (!healthTracker.isStorageFull()) {
                             log.warn("{} STORAGE FULL. ENTERING READ ONLY MODE. Any non-critical writes will be rejected.", logPrefix);
                         }
                         log.warn("{} STORAGE FULL - used={} total={}.", logPrefix, used, config.getMaxSafeStorageSize());
                     } else {
-                        if (isStorageFull.get()) {
+                        if (healthTracker.isStorageFull()) {
```

Comment:
Doesnt the SS container crash and restart on storage becoming full ?? (remember seeing it on one of the setups). Will we see this transition of "restoring normal writes" if it crashes?

Reply:
Concept of "soft quota" was added earlier to prevent sudden crash/unavailability etc.
Storage full here is a soft quota meaning the storage usage is above set soft threshold. We still allow writes to any segments under _system folder